src/metabase/driver/mysql/actions.clj

```diff
@@ -137,9 +137,11 @@
   (let [schema (when (next table-components) (first table-components))
         table (last table-components)]
     (jdbc/with-db-metadata [md db-spec]
-      (->> (.getPrimaryKeys md nil schema table)
-           jdbc/metadata-result
-           (mapv :column_name)))))
+      (with-open [rset (.getPrimaryKeys md nil schema table)]
+        (loop [acc []]
```

Comment:
Dan, you're right that this would be slightly faster with a transient vector, but I don't think it's really an important difference -- avoiding all the maps that `clojure.java.jdbc` would have created for us that we immediately tossed out will be the real win here